config/config_test.go

```diff
@@ -183,191 +221,66 @@ func TestGetOptionalParameters(t *testing.T) {
-		cfgService := makeCfgService(true, false)
-		cfgService.Attributes["data_dir"] = true
-		_, err := newConfig(cfgService)
-		expE := newError("1 error(s) decoding:\n\n* 'data_dir' expected type 'string', got unconvertible type 'bool', value: 'true'")
-		test.That(t, err, test.ShouldBeError, expE)
-		cfgService.Attributes["data_dir"] = "true"
-		_, err = newConfig(cfgService)
-		test.That(t, err, test.ShouldBeNil)
-	})
-
-	t.Run("Config with invalid camera data_frequency_hz with feature flag enabled", func(t *testing.T) {
-		cfgService := makeCfgService(true, false)
-		cfgService.Attributes["camera"] = map[string]string{
-			"name":              "a",
-			"data_frequency_hz": "twenty",
-		}
-		_, err := newConfig(cfgService)
-		test.That(t, err, test.ShouldBeError, newError("camera[data_frequency_hz] must only contain digits"))
-	})
-
-	t.Run("Config with out of range values with feature flag enabled", func(t *testing.T) {
-		cfgService := makeCfgService(true, false)
-		cfgService.Attributes["camera"] = map[string]string{
-			"name":              "a",
-			"data_frequency_hz": "-1",
-		}
-		_, err := newConfig(cfgService)
-		test.That(t, err, test.ShouldBeError, newError("cannot specify camera[data_frequency_hz] less than zero"))
-		cfgService.Attributes["camera"] = map[string]string{
-			"name":              "a",
-			"data_frequency_hz": "1",
+		if imuIntegrationEnabled {
+			test.That(t, imuName, test.ShouldEqual, "testNameSensor")
+			test.That(t, imuDataRateMsec, test.ShouldEqual, 500)
+			test.That(t, lidarDataRateMsec, test.ShouldEqual, 500)
+		} else {
+			test.That(t, lidarDataRateMsec, test.ShouldEqual, 50)
 		}
-		cfgService.Attributes["map_rate_sec"] = -1
-		_, err = newConfig(cfgService)
-		test.That(t, err, test.ShouldBeError, newError("cannot specify map_rate_sec less than zero"))
+		test.That(t, enableMapping, test.ShouldBeFalse)
 	})
 
-	t.Run("All parameters e2e with feature flag enabled", func(t *testing.T) {
-		cfgService := makeCfgService(true, false)
-		cfgService.Attributes["camera"] = map[string]string{
-			"name":              "a",
-			"data_frequency_hz": "20",
-		}
-		cfgService.Attributes["map_rate_sec"] = 1002
-
-		cfgService.Attributes["config_params"] = map[string]string{
-			"mode":    "test mode",
-			"value":   "0",
-			"value_2": "test",
-		}
-		cfg, err := newConfig(cfgService)
-		test.That(t, err, test.ShouldBeNil)
-		test.That(t, cfg.DataDirectory, test.ShouldEqual, cfgService.Attributes["data_dir"])
-		test.That(t, cfg.Camera, test.ShouldResemble, cfgService.Attributes["camera"])
-		test.That(t, *cfg.MapRateSec, test.ShouldEqual, cfgService.Attributes["map_rate_sec"])
-		test.That(t, cfg.ConfigParams, test.ShouldResemble, cfgService.Attributes["config_params"])
-	})
+	if imuIntegrationEnabled {
+		sensorAttributeTestHelper(t, logger, imuIntegrationEnabled, cloudStoryEnabled)
+	}
 }
 
-func TestGetOptionalParametersFeatureFlag(t *testing.T) {
```

Comment:
this is all tested now using the newly defined helper ` getOptionalParametersTestHelper`